src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -177,15 +187,16 @@ ValidatorList::load(
             JLOG(j_.warn()) << "Duplicate node identity: " << match[1];
             continue;
         }
-        auto it = publisherLists_.emplace(
+        auto [it, inserted] = publisherLists_.emplace(
```

Comment:
I have a few observations - it seems like the variable `local` is unnecessary as it isn't being modified to reflect information from `configKeys`. Additionally, if more than one validator is listed under the `[validators]` config section, only one entry will be added to `publisherLists_` because of the way `local` is being used. In any case, the entry added to `publisherLists_` won't reflect the key(s) present in the `validators` config section.

Reply:
You're partially right - I was able to get rid of `local` entirely, and `emplace` default objects. The reason it's not modified is because it is used as a "pseudo master key" for a UNL _signer_, not a validator. The validator keys are iterated with `n`, which is used to make `id`. On repeated calls, the `emplace` function simply returns the existing entry. So the first call uses a default `PublicKey`, and creates a new `PublisherListCollection`. Then `id` is added to the `PublisherListCollection`'s current list. On later calls, `emplace` returns the same `PublisherListCollection` that was created the first time, so `id` is added to the list. At the end of the process, the `list` contains all the `PublicKeys` corresponding to `configKeys`. You can look at `testConfigLoad` in ValidatorList_test.cpp to see this working in action.